### PR TITLE
Migrate structopt to clap v3

### PR DESCRIPTION
**Pull request by bisgardo**
## Purpose

As `clap` v3 is on par with or better than `structopt`, that project [has entered maintenance mode](https://github.com/TeXitoi/structopt#maintenance) and will not have things like the weird way it handles env vars fixed.

## Changes

Upgraded `clap` to v. `3.1` and removed dependency on `structopt`.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -13,42 +13,47 @@ use crate::{
...
-    #[structopt(
+#[derive(Parser, Debug)]
+#[clap(
+    author = "Concordium Foundation",
+    about = "A server implementing the Rosetta API for the Concordium blockchain."
+)]
+struct Args {
+    #[clap(
         long = "network",
+        env = "NETWORK",
```
Comment (abizjak) [inline src/main.rs:29]: I think in general it is better to not use such generic environment variables as `NETWORK` and `PORT`.

In other projects we have we follow the convention to have environment variables `BINARY_NAME_...` or something close to `BINARY_NAME`.

**Review by abizjak — APPROVED** [review]
I think upgrading to clap 3.1 is good, although I would have preferred to do that once for all projects since the API is quite different (in details, e.g., defaults, behaviour) between the versions.

But  I see that it is a good time to do it now for rosetta.

I do have a general concern about the naming of environment variables. They are too generic and might be set accidentally leading to unexpected behaviour. But that is not a subject of this PR.

File: src/main.rs (hunk 1 above)
Comment (bisgardo) [reply src/main.rs:29]: Done in https://github.com/Concordium/concordium-rosetta/pull/7.

**Comment by bisgardo** [comment]
> Although it also looks like we cannot use clap 3.1 with rust 1.53, so we need to wait with upgrading until we upgrade our toolchains to 1.54+.

Building and running with 1.54 works just fine. I'm fine with not merging this until we're upgrading the rest of the toolchain, but this could also serve as a guineapig for the next rust version?

**Comment by bisgardo** [comment]
@abizjak do you think it's ok to merge now with the bump to 1.54?

**Comment by abizjak** [comment]
> @abizjak do you think it's ok to merge now with the bump to 1.54?

It's not ideal, but if clippy does not produce any new warnings with 1.54 then it is the better option. We'll soon be upgrading the Rust toolchain so it does not make sense to hold this back since it's a new project.

**Comment by bisgardo** [comment]
> > @abizjak do you think it's ok to merge now with the bump to 1.54?
> 
> It's not ideal, but if clippy does not produce any new warnings with 1.54 then it is the better option. We'll soon be upgrading the Rust toolchain so it does not make sense to hold this back since it's a new project.

Clippy is happy; merging.

**Pull request merged**